src/types.rs

```diff
@@ -0,0 +1 @@
+pub mod types;
```

Comment:
Now, after seeing what's in the file, I think calling it `types.rs` is a bit too broad. How about something like `field_element.rs`? Depends on what will end up here eventually, but for now it seems closely related specifically to the Poseidon hash.

Also, I now see that you probably modeled it after my `utils.rs`, which contains `storage.rs`. I did that because I expect many different utilities to end up in there. In your case, I think I'd start without such nested structure because it's not sure what else will really fit in there.